desc/experimental/_terpsichore.py

```diff
@@ -0,0 +1,872 @@
+    stop_time=60,
+    data_transforms=None,
+    fit_transform=None,
+    theta0_outboard=True,
+):
+    """TERPSICHORE driver function."""
+    # TERPSICHORE assumes theta=0 is on the outboard midplane
+    if not theta0_outboard:
+        eq = flip_theta(eq)
+
+    # create temporary directory to store I/O files
+    tmp_path = os.path.join(path, "tmp-TERPS")
+    os.mkdir(tmp_path)
+
+    # write input files for each equilibrium in serial
+    idxs = list(range(len(eq)))  # equilibrium indices
+    for k in idxs:
+        idx_path = os.path.join(tmp_path, str(k))
+        os.mkdir(idx_path)
+        exec_path = os.path.join(idx_path, exec)
+        input_path = os.path.join(idx_path, "input")
+        wout_path = os.path.join(idx_path, "wout.nc")
+        shutil.copy(os.path.join(path, exec), exec_path)
+        _write_wout(
+            eq=eq[k],
+            path=wout_path,
+            surfs=surfs,
+            data_transforms=data_transforms,
+            fit_transform=fit_transform,
+        )
+        _write_terps_input(
+            path=input_path,
+            mode_family=mode_family,
+            surfs=surfs,
+            lssl=lssl,
+            lssd=lssd,
+            M_max=M_max,
+            N_min=N_min,
+            N_max=N_max,
+            M_booz_max=M_booz_max,
+            N_booz_max=N_booz_max,
+            awall=awall,
+            deltaJp=deltaJp,
+            modelk=modelk,
+            nfp=eq[k].NFP,
+            nev=nev,
+            al0=al0,
+        )
+
+    # run TERPSICHORE on list of equilibria in parallel
+    if len(eq) == 1:  # no multiprocessing if only one equilibrium
+        result = jnp.atleast_1d(
+            _pool_fun(
+                0,
+                path=tmp_path,
+                exec=exec,
+                scalar=scalar,
+                surfs=surfs,
+                al0=al0,
+                sleep_time=sleep_time,
+                stop_time=stop_time,
+            )
+        )
+    else:
+        with mp.Pool(processes=min(processes, len(eq))) as pool:
+            results = pool.map(
+                functools.partial(
+                    _pool_fun,
+                    path=tmp_path,
+                    exec=exec,
+                    scalar=scalar,
+                    surfs=surfs,
+                    al0=al0,
+                    sleep_time=sleep_time,
+                    stop_time=stop_time,
+                ),
+                idxs,
+            )
+            pool.close()
+            pool.join()
+            result = jnp.vstack(results, dtype=float)
+
+    # remove temporary directory
+    shutil.rmtree(tmp_path)
+
+    return result
+
+
+def _write_wout(eq, path, surfs, data_transforms, fit_transform):  # noqa: C901
+    """Write the wout NetCDF file from the equilibrium."""
+    # this is a lightweight version of VMECIO.save
+    file = Dataset(path, mode="w", format="NETCDF3_64BIT_OFFSET")
+
+    Psi = eq.Psi
+    NFP = eq.NFP
+    M = eq.M
+    N = eq.N
+    M_nyq = M + 4
```

Comment:
these are the default values that VMEC has, but tbh it is probably better to increase these from their defaults or add an option to the objective to pass in the nyquist resolutions if desired. 

I doubt that M+4 is sufficient to describe sqrt(g) even, as that is a product of 3 Fourier series of M,N each. Might be worth a convergence test at differing nyquist resolutions to see how much of an effect this has

Reply:
and in VMEC, if the eq was solved with NTHETA and NZETA set, that is what sets the nyquist resolutions, so thuogh M+4 and N+2 are the defaults, people may be using higher values when they know they are gonna pass to Terpsichore, or if they know they need higher resolution for stuff like free bdry.

Reply:
Update this to reflect the changes made in #1543